src/Nethermind/Nethermind.Merge.Plugin/Handlers/GetPayloadBodiesByHashV2Handler.cs

```diff
@@ -0,0 +1,75 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Nethermind.Blockchain;
+using Nethermind.Core;
+using Nethermind.Core.ConsensusRequests;
+using Nethermind.Core.Crypto;
+using Nethermind.JsonRpc;
+using Nethermind.Logging;
+using Nethermind.Merge.Plugin.Data;
+
+namespace Nethermind.Merge.Plugin.Handlers;
+
+
+public class GetPayloadBodiesByHashV2Handler(IBlockTree blockTree, ILogManager logManager) : GetPayloadBodiesByHashV1Handler(blockTree, logManager), IAsyncHandler<IList<Hash256>, IEnumerable<ExecutionPayloadBodyV2Result?>>
+{
+    public new Task<ResultWrapper<IEnumerable<ExecutionPayloadBodyV2Result?>>> HandleAsync(IList<Hash256> blockHashes)
+    {
+        if (blockHashes.Count > MaxCount)
+        {
+            var error = $"The number of requested bodies must not exceed {MaxCount}";
+
+            if (_logger.IsError) _logger.Error($"{nameof(GetPayloadBodiesByHashV2Handler)}: {error}");
+
+            return ResultWrapper<IEnumerable<ExecutionPayloadBodyV2Result?>>.Fail(error, MergeErrorCodes.TooLargeRequest);
+        }
+
+        return Task.FromResult(ResultWrapper<IEnumerable<ExecutionPayloadBodyV2Result?>>.Success(GetRequests(blockHashes)));
+    }
+
+    private IEnumerable<ExecutionPayloadBodyV2Result?> GetRequests(IList<Hash256> blockHashes)
+    {
+        for (int i = 0; i < blockHashes.Count; i++)
+        {
+            Block? block = _blockTree.FindBlock(blockHashes[i]);
+
+            ConsensusRequest[]? consensusRequests = block?.Requests;
+
+            List<Deposit> deposits = [];
```

Comment:
You can do it without lists, chec ExecutionPayloadV4, pasted the code below

```
            (int depositCount, int withdrawalRequestCount) = blockRequests.GetTypeCounts();
            executionPayload.DepositRequests = new Deposit[depositCount];
            executionPayload.WithdrawalRequests = new WithdrawalRequest[withdrawalRequestCount];
            int depositIndex = 0;
            int withdrawalRequestIndex = 0;
            for (int i = 0; i < blockRequests.Length; ++i)
            {
                ConsensusRequest request = blockRequests[i];
                if (request.Type == ConsensusRequestsType.Deposit)
                {
                    executionPayload.DepositRequests[depositIndex++] = (Deposit)request;
                }
                else
                {
                    executionPayload.WithdrawalRequests[withdrawalRequestIndex++] = (WithdrawalRequest)request;
                }
            }
```

Reply:
we are expected to return `ExecutionPayloadBodyV2Result` object.  And, `executionPayload` is used in the above snippet